### PR TITLE
Use env vars to customize backup retention

### LINKED ISSUE #225: Default backup retention period
Good day.

We use etcd-manager with kops to manage etcd. By default, etcd-managetr sets up backups to the bake, every 15 minutes. But I could not find out what default retention (https://github.com/kopeio/etcd-manager/blob/master/pkg/backupcontroller/cleanup.go) is worth it and how can it be configured?

### LINKED ISSUE #308: Remove old backups
Hello,

It's safe create a s3 lifecycle policy to delete old etcd backups (s3://BUCKET/kops/backups/etcd/main/ and s3://BUCKET/kops/backups/etcd/events/)? I can't found any etcd-backup-manger config to do this.

Thank you

### PR DESCRIPTION
**Pull request by hakman**
At the moment backup retention is hardcoded.
Hourly backups are kept for 7 days and daily backups are kept for 1 year.

Env vars could be used to customise backup retention:
```
ETCD_MANAGER_HOURLY_BACKUPS_RETENTION=7d
ETCD_MANAGER_DAILY_BACKUPS_RETENTION=1y
```

Fixes #225 
Fixes #308

### CODE REVIEW

File: pkg/backupcontroller/cleanup.go [hunk 1]
```diff
@@ -3,13 +3,66 @@ package backupcontroller
...
+// HourlyBackupsRetention controls how long hourly backups are kept.
+var HourlyBackupsRetention = 24 * 7 * time.Hour
+
+// DailyBackupsRetention controls how long daily backups are kept.
+var DailyBackupsRetention = 24 * 7 * 365 * time.Hour
+
+// ParseHumanDuration parses a go-style duration string, but
+// recognizes additional suffixes: d means "day" and is interpreted as
+// 24 hours; y means "year" and is interpreted as 365 days.
+func ParseHumanDuration(s string) (time.Duration, error) {
```
Comment (hakman) [inline pkg/backupcontroller/cleanup.go:25]: This was shamelessly copied from `certs.go`. I couldn't find some good common utils place for it.
https://github.com/kopeio/etcd-manager/blob/84bd90884129b71ff5aae0aa36e88ab196dd2af5/pkg/pki/certs.go#L32

**Comment by johngmyers** [comment]
Why would we use env vars instead of flags or a config file?

**Comment by hakman** [comment]
> Why would we use env vars instead of flags or a config file?

No special reason, just followed the approach from https://github.com/kopeio/etcd-manager/commit/f4b782c28f3d8e7df88e65dffb4514d366b0cc5a.

**Comment by justinsb** [comment]
I wouldn't mind also adding flags, but let's get this going as-is.  If we start specifying these values directly in the etcd struct in kops, we'll have to deal with the fact that they can be specified in both env and directly, but we have to deal with that anyway for the metrics-url.  (cc @granular-ryanbonham )

Thanks for this @hakman !

/approve
/lgtm

**Comment by caiohasouza** [comment]
Thank you @hakman, great job!

**Pull request merged**